packages/block-library/src/code/index.js

```diff
@@ -68,6 +73,58 @@ export const settings = {
 	edit,
 
 	save( { attributes } ) {
-		return <pre><code>{ attributes.content }</code></pre>;
+		const content = flow(
+			escapeAmpersands,
+			escapeOpeningSquareBrackets,
+			escapeProtocolInIsolatedUrls
+		)( attributes.content || '' );
+		return <pre><code>{ content }</code></pre>;
 	},
 };
+
+/**
+ * Returns the given content with all its ampersand characters converted
+ * into their HTML entity counterpart (i.e. & => &amp;)
+ *
+ * @param {string}  content The content of a code block.
+ * @return {string} The given content with its ampersands converted into
+ *                  their HTML entity counterpart (i.e. & => &amp;)
+ */
+function escapeAmpersands( content ) {
+	return content.replace( /&/g, '&amp;' );
+}
+
+/**
+ * Returns the given content with all opening shortcode characters converted
+ * into their HTML entity counterpart (i.e. [ => &#91;). For instance, a
+ * shortcode like [embed] becomes &#91;embed]
+ *
+ * This function replicates the escaping of HTML tags, where a tag like
+ * <strong> becomes &lt;strong>.
+ *
+ * @param {string}  content The content of a code block.
+ * @return {string} The given content with its opening shortcode characters
+ *                  converted into their HTML entity counterpart
+ *                  (i.e. [ => &#91;)
+ */
+function escapeOpeningSquareBrackets( content ) {
+	return content.replace( /\[/g, '&#91;' );
+}
+
+/**
+ * Converts the first two forward slashes of any isolated URL into their HTML
+ * counterparts (i.e. // => &#47;&#47;). For instance, https://youtube.com/watch?x
+ * becomes https:&#47;&#47;youtube.com/watch?x.
+ *
+ * An isolated URL is a URL that sits in its own line, surrounded only by spacing
+ * characters.
+ *
+ * See wp-includes/class-wp-embed.php:387 (function autoembed)
```

Comment:
Line numbers without an associated version/tag/commit are very likely to fall out of sync as future changes are made to the file. In fact, in the current trunk version it currently points to a line which has no content:

https://github.com/WordPress/wordpress-develop/tree/master/src/wp-includes/class-wp-embed.php#L387

A link like this may work better, as it's pinned to a specific version:

https://github.com/WordPress/wordpress-develop/blob/5.0.3/src/wp-includes/class-wp-embed.php#L387

Reply:
Fixed in c2a352cdaabf7b7d5ffe4dd71fdd72220e00feaf